docs/rules/group-exports.md

```diff
@@ -0,0 +1,62 @@
+# group-exports
+
+Reports when multiple named exports or CommonJS assignments are present in a single file and when the default export is not adjacent to the named export.
+
+**Rationale:** An `export` declaration or `module.exports` assignment can appear anywhere in the code. By requiring a single export declaration along with the default export being placed next to the named export, your exports will remain at one place, making it easier to see what exports a module provides.
+
+## Rule Details
+
+This rule warns whenever a single file contains multiple named exports or assignments to `module.exports` (or `exports`) and when the default export is not adjacent to the named export.
+
+### Valid
+
+```js
+// Default export statement is adjacent to named export -> ok
+export default function test() {}
+// A single named export statement -> ok
+export const valid = true
+```
+
+```js
+const first = true
+const second = true
+
+// A single named export statement -> ok
+export {
+  first,
+  second,
+}
+```
+
+```js
+// A single exports assignment -> ok
+module.exports = {
+  first: true,
+  second: true
+}
+```
+
+### Invalid
+
+```js
+// Multiple named export statements -> not ok!
+export const first = true
+export const second = true
+```
+
+```js
+// Default export is not adjacent to the named export statement -> not ok!
+export default {}
+const first = true
+export { first }
+```
+
+```js
+// Multiple exports assignments -> not ok!
+exports.first = true
+exports.second = true
```

Comment:
@robertrossmann yes, i'm aware of that; it could still be invalid, but doesn't necessarily have to be covered by this rule.

imo `exports.foo = {}` should be treated the same as `module.exports.foo = {}`.

If someone wants to export a function with extra properties, they should do something like `const foo = () => {}; foo.first = true; foo.second = true; module.exports = foo;` - I don't think the rule should allow multiple module.exports mutations under any circumstances.